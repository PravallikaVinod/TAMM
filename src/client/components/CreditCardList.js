import React from  'react';
import {GlobalStyle,SubTitle1} from '../styles/style.js';

class CreditCardList extends React.Component {

    constructor(props){
        super(props)
    }

render(){ 
    let  renderList = "";
    if(this.props.cardsList != undefined){
    renderList =   this.props.cardsList.map(function(data,index){
            return( <tr key={index}>
                 <td>{data.name}</td>
                 <td>{data.cardNumber}</td>
                 <td>{data.balance}</td>
                 <td>{data.limit}</td>
             </tr>);
             })
     }
  
return(
<div>
<div><SubTitle1>Existing Cards</SubTitle1> </div>
<GlobalStyle/>
<table>
<thead>
<tr>
<th> Name </th>
<th> Card Number </th>
<th> Balance </th>
<th> Limit </th>
</tr>
</thead>
<tbody>
{renderList}
</tbody>

</table>
</div>
);
    }
}


export default CreditCardList;