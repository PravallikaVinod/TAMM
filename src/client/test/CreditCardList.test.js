import React from  'react';
import {shallow,render,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CreditCardList from '../components/CreditCardList.js';

import { configure } from 'enzyme';

configure({ adapter: new Adapter() });


describe('Credit Card List Component ', () => {

    let list = [];
    list.name="Sita";
    list.cardNumber = 79927398713;
    list.limit = 100000;
    list.balance = 0;
    const wrapper = mount(<CreditCardList cardsList = {list} />);
    it('renders correctly', () => {   
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a `<table>` element', () => {
        expect(wrapper.find('table').length).toBe(1);
    });

});