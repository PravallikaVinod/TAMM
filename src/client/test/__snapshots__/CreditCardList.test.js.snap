// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit Card List Component  renders correctly 1`] = `ReactWrapper {}`;
